airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -254,15 +211,29 @@ public static void runServer(final Set<ContainerRequestFilter> requestFilters,
 
     if (airbyteDatabaseVersion.isPresent() && AirbyteVersion.isCompatible(airbyteVersion, airbyteDatabaseVersion.get())) {
       LOGGER.info("Starting server...");
-      new ServerApp(configRepository, jobPersistence, configs, requestFilters, responseFilters).start();
+
+      final JobTracker jobTracker = new JobTracker(configRepository, jobPersistence);
+      final WorkflowServiceStubs temporalService = TemporalUtils.createTemporalService(configs.getTemporalHost());
+      final TemporalClient temporalClient = TemporalClient.production(configs.getTemporalHost(), configs.getWorkspaceRoot());
+      final SchedulerJobClient schedulerJobClient = new DefaultSchedulerJobClient(jobPersistence, new DefaultJobCreator(jobPersistence));
+      final DefaultSynchronousSchedulerClient syncSchedulerClient = new DefaultSynchronousSchedulerClient(temporalClient, jobTracker);
+      final SpecCachingSynchronousSchedulerClient cachingSchedulerClient = new SpecCachingSynchronousSchedulerClient(syncSchedulerClient);
+
+      return apiFactory.create(
```

Comment:
What about simplifying this constructor by passing in only `configRepository`, `jobPersistence` and `configs`? L214 - 222 can be pushed into the  `ServerFactory`.

Reply:
You can't reliably create these clients based off of mocks() if they have validations internally, which makes it much harder to inject. I'm planning on using this functionality in tests soon.

Reply:
Fair. I think I was thinking of two constructors,  a slimmer down one for production and an exposed one for testing. We can bolt this on later.